core/go/internal/keymanager/key_resolver.go

```diff
@@ -233,8 +233,15 @@ func (kr *keyResolver) resolveKey(ctx context.Context, identifier, algorithm, ve
 	kr.l.Lock()
 	defer kr.l.Unlock()
 
-	// Identifier must be a valid
-	if err := pldtypes.ValidateSafeCharsStartEndAlphaNum(ctx, identifier, pldtypes.DefaultNameMaxLen, "identifier"); err != nil {
+	// Identifier must be valid and not an eth address
```

Comment:
@annamcallister - can you share the thinking here?

Not sure I disagree with it, but it seems very architecturally significant on the signing side (unrelated to the topic).

e.g. it seems related to if I want to directly use the `0x` address of my keys to resolve them, and the chain of configuration between my application, and my backend storage system (file system with key files... or advanced plugged in key manager system with keys stored by `name` like AWS/Azure/GCP Cloud HSMs etc.).